digits/extensions/view/imageSegmentation/view.py

```diff
@@ -134,24 +171,68 @@ def process_data(self, input_id, input_data, output_data):
         """
         # assume the only output is a CHW image where C is the number
         # of classes, H and W are the height and width of the image
-        data = output_data[output_data.keys()[0]].astype('float32')
+        class_data = output_data[output_data.keys()[0]].astype('float32')
         # retain only the top class for each pixel
-        data = np.argmax(data,axis=0)
+        class_data = np.argmax(class_data, axis=0).astype('uint8')
 
         # remember the classes we found
-        found_classes = np.unique(data)
+        found_classes = np.unique(class_data)
         self.found_classes = np.unique(np.concatenate(
             (self.found_classes, found_classes)))
 
         # convert using color map (assume 8-bit output)
         if self.map:
-            data = self.map.to_rgba(data)*255
-            # keep RGB values only, remove alpha channel
-            data = data[:, :, 0:3]
-
-        # convert to uint8
-        data = data.astype('uint8')
-        # convert to PIL image
-        image = PIL.Image.fromarray(data)
-
-        return image
+            fill_data = (self.map.to_rgba(class_data)*255).astype('uint8')
+        else:
+            fill_data = np.ndarray((class_data.shape[0], class_data.shape[1], 4), dtype='uint8')
+            for x in xrange(3):
+                fill_data[:, :, x] = class_data.copy()
+
+        zeros = np.zeros(class_data.shape)
+        # Assuming that class 0 is the background
+        mask = np.greater(class_data, zeros)
+        fill_data[:, :, 3] = mask * 255
+
+        # Black mask of non-segmented pixels
+        mask_data = np.zeros(fill_data.shape, dtype='uint8')
```

Comment:
The np.zeros create RGBA data. the line below sets the A component. Is there some pythonic trickery that I'm missing?